go/arrow/ipc/file_reader.go

```diff
@@ -348,10 +356,26 @@ func (src *ipcSource) buffer(i int) *memory.Buffer {
 		return memory.NewBufferBytes(nil)
 	}
 
-	raw := make([]byte, buf.Length())
-	_, err := src.r.ReadAt(raw, buf.Offset())
-	if err != nil {
-		panic(err)
+	var raw []byte
+	if src.codec == nil {
+		raw = make([]byte, buf.Length())
+		_, err := src.r.ReadAt(raw, buf.Offset())
+		if err != nil {
+			panic(err)
+		}
+	} else {
+		sr := io.NewSectionReader(src.r, buf.Offset(), buf.Length())
+		var uncompressedSize uint64
+		if err := binary.Read(sr, binary.LittleEndian, &uncompressedSize); err != nil {
```

Comment:
I think this needs to check for the magic uncompressed value as well and revert back to a slice? Are zero length buffers handled upstream from this code?

Reply:
zero length buffers are handled above this in this function, if the buffer is length 0, this function returns a buffer with a nil slice. In addition the writer doesn't attempt to compress or write for a 0 length buffer either.